client/windows/CCastleInterface.cpp

```diff
@@ -1891,7 +1891,7 @@ const CBuilding * CFortScreen::RecruitArea::getMyBuilding()
 	{
 		if (town->hasBuilt(myID))
 			build = town->town->buildings.at(myID);
-		myID.advance(GameConstants::CREATURES_PER_TOWN);
+		myID.advance(town->town->creatures.size());
```

Comment:
I think this would break upgraded buildings in fort?
Can you check that Fort window works correctly with upgraded buildings, both for tiers 1-7 and for tier 8?
Because for tiers 1-7 advancing needs to be done by 7 (ID of upgraded dwelling - ID of downgrade) and for new tier 8 - only by 1.

I think this needs your new `getDwellingFromLevel` method?

Reply:
Fort is absolutely fine for my. With 7 and 8 creatures.
Tested in Fort using vcmibuild (also tested with "empty" town):
- Buying (and selecting other level)
- Right click